src/PQueue.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+
+export default class PQueue {
+
+  constructor() {
+    this._list = []
+    this._backElementIndex = -1
+  }
+
+  enqueue( element, priority ) {
+    if ( priority < 1 ) {
+      throw new Error( 'This priority is invalid' )
+    }
+    this._list[ ++this._backElementIndex ] = { element, priority }
+    this._list = this._list.sort( ( element, nextElement ) => {
+      return element.priority > nextElement.priority
+    } )
+  }
+
+  dequeue() {
+    let highestPriorityElement = this.front()
+    this._list = this._list.reduce( ( memo, element, index ) => {
```

Comment:
Please, please, please use whitespace in your code.  Within a function, I recommend whitespace between setup, body, and return statement at least, in order to increase readability.